File: server/src/main/java/com/edutech/progressive/entity/Supplier.java
package com.edutech.progressive.entity;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Supplier implements Comparable<Supplier> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int supplierId;
    private String supplierName;
    private String email;
    private String phone;
    private String address;
    private String username;
    private String password;
    private String role;

    public Supplier() {
    }

    public Supplier(int supplierId, String supplierName, String email, String phone, String address, String username, String password, String role) {
        this.supplierId = supplierId;
        this.supplierName = supplierName;
        this.email = email;
        this.phone = phone;
        this.address = address;
        this.username = username;
        this.password = password;
        this.role = role;
    }

    public String getEmail() {
        return email;
    }

    public int getSupplierId() {
        return supplierId;
    }

    public void setSupplierId(int supplierId) {
        this.supplierId = supplierId;
    }

    public String getSupplierName() {
        return supplierName;
    }

    public void setSupplierName(String supplierName) {
        this.supplierName = supplierName;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    @Override
    public int compareTo(Supplier otherSupplier) {
        // Implement comparison logic based on account balance
        return this.getSupplierName().compareTo(otherSupplier.getSupplierName());
    }
}

File: client/src/app/supplylink/types/Supplier.ts
export class Supplier {
  supplierId: number | undefined;
  supplierName: string;
  email: string;
  phone: string;
  address: string;
  username: string;
  password: string;
  role?: string;

  constructor(
    supplierId: number | undefined,
    supplierName: string,
    email: string,
    phone: string,
    address: string,
    username: string,
    password: string,
    role?: string
  ) {
    this.supplierId = supplierId;
    this.supplierName = supplierName;
    this.email = email;
    this.phone = phone;
    this.address = address;
    this.username = username;
    this.password = password;
    this.role = role;
  }

  // displayInfo() {
  //   // Each log must be ONE formatted string with exact labels/casing
  //   console.log(`Supplier ID: ${this.supplierId}`);
  //   console.log(`Supplier name: ${this.supplierName}`);
  //   console.log(`email: ${this.email}`);
  // }
}